### ISSUE TITLE
Issue with math lib

### ISSUE THREAD

**Issue by ThirteenAG** [ps2dev/ps2sdk#255]
Hey, I'm using this sdk to compile a plugin for a game, but there's a strange issue with math operations that I can't quite figure out.
I made a minimal example of a code that does something in the game, it seems working and produces the following look:
![pcsx2x64_2022-04-30_17-16-56](https://user-images.githubusercontent.com/4904157/166109474-d7949c26-b72c-4653-80c2-268c10476021.png)
![pcsx2x64_2022-04-30_17-17-11](https://user-images.githubusercontent.com/4904157/166109476-aeb171ca-0eb1-4d47-a36b-024fcd805df0.png)

That's the goal, it's how I intend it to look. Next, I have an unrelated function in the code, that is unused:
```c
float AdjustFOV(float f, float ar, float t)
{
    return ((2.0f * (((ar) / (4.0f / 3.0f)) * t)) * (180.0f / (float)M_PI) * 100.0f) / 100.0f;
}
```
That way it looks in the code and when compiled, on screenshots above. Let me remind that it's entirely unused.

Next, I'm adding atanf in the code:
```c
float AdjustFOV(float f, float ar, float t)
{
    return ((2.0f * atanf(((ar) / (4.0f / 3.0f)) * t)) * (180.0f / (float)M_PI) * 100.0f) / 100.0f;
}
```

And despite the fact that this code is never called, this happens ingame:

![pcsx2x64_2022-04-30_17-17-42](https://user-images.githubusercontent.com/4904157/166109561-ff34137f-1d78-4989-83b2-87f60014183b.png)
![pcsx2x64_2022-04-30_17-17-48](https://user-images.githubusercontent.com/4904157/166109559-c485427e-fdc3-4a62-b582-89f6942227c3.png)

Can someone explain why it happens? Doesn't make any sense to me. Project compiled with `EE_LIBS += -l:libc.a -l:libm.a -l:libgcc.a`

**Comment by uyjulian** [comment]
eabi/n32 mismatch issue

This doesn't seem to be an issue anymore. Reopen if this is not the case

**Comment by ThirteenAG** [comment]
Thanks for letting me know.

I build on windows, so it's a bit cumbersome to upgrade:
`$(DIR_GUARD)` doesn't work on windows, had to disable it.
Dll dependencies are completely missing for portable build system, I add them manually, and had to add libmpfr-6.dll and libisl-23.dll(renamed libisl-15.dll). Would be cool to bundle them in the archive.

Anyway, back to this issue:
I have this func:
```c
float (*game_atan2f)(float a1, float a2);
float AdjustFOV(float f, float ar, float t)
{
    return ((2.0f * game_atan2f(((ar) / (4.0f / 3.0f)) * t, 1.0f)) * (180.0f / (float)M_PI) * 100.0f) / 100.0f;
}
```
To test, I compiled my plugin as is, everything seems normal:
![pcsx2x64_2023_12_01_20_51_50](https://github.com/ps2dev/ps2sdk/assets/4904157/879b1e88-dda4-4209-9760-e89f555cbb39)
Then I replaced `game_atan2f` to just `atan2f`, and there is a problem:
![pcsx2x64_2023_12_01_20_51_12](https://github.com/ps2dev/ps2sdk/assets/4904157/9661a397-c07a-4d33-ab8f-0f52d981135f)

Headlights are bigger than they should be, this function doesn't affect them in any way though, I guess some kind of corruption somewhere. I didn't notice any other issues, and FOV is calculated correctly.
